oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -340,6 +343,23 @@ public AccessToken refreshAccessToken() throws IOException {
     HttpRequest request = requestFactory.buildPostRequest(new GenericUrl(tokenServerUri), content);
     request.setParser(new JsonObjectParser(jsonFactory));
 
+    request.setIOExceptionHandler(new HttpBackOffIOExceptionHandler(new ExponentialBackOff()));
+    request.setUnsuccessfulResponseHandler(
+        new HttpBackOffUnsuccessfulResponseHandler(new ExponentialBackOff()).setBackOffRequired(
+            new BackOffRequired() {
+              public boolean isRequired(HttpResponse response) {
+                int code = response.getStatusCode();
+                return (
+                    // Server error --- includes timeout errors, which use 500 instead of 408
+                    code / 100 == 5
```

Comment:
This seems like an unnecessary trick. Taking into account that there is nothing above server errors in http standard (no 600+ error codes), please consider changing it to `code >= 500`